doc/specs/vulkan/chapters/pipelines.txt

```diff
@@ -617,7 +617,8 @@ endif::VK_KHR_maintenance2[]
     If no element of the pname:pDynamicStates member of pname:pDynamicState
     is ename:VK_DYNAMIC_STATE_VIEWPORT, the pname:pViewports member of
     pname:pViewportState must: be a valid pointer to an array of
-    pname:pViewportState::pname:viewportCount sname:VkViewport structures
+    pname:pViewportState::pname:viewportCount valid sname:VkViewport
```

Comment:
VkViewport doesn't have an sType. Is is appropriate to to refer to "valid structures" without an sType? What makes it a valid structure?
There's a separate VU for "vkCmdSetViewport()" that refers to "valid VkViewport structures" and I agree that these two should be consistent. Just not sure "valid * structure" makes sense for structs w/o sTypes and reviewing "Valid Usage for Structure*" sections of [spec](https://www.khronos.org/registry/vulkan/specs/1.0-extensions/html/vkspec.html#fundamentals-validusage-sType) does not shed any light on how a non-sType struct could/would be "valid."

Reply:
Yes. You are looking at wrong chapter of implicit valid usage of `sType`. Look here:

> Valid usage conditions are described in a block labelled “Valid Usage” following **each** command or **structure** they apply to.

`VkViewport` has Valid Usage section and therefore needs to be said it has to be "valid".
Lack of "valid" means that structure either is returned only in that role, or it is guaranteed to be always valid (i.e. has no Valid Usage section; be it implicit or explicit).

Reply:
> PS BTW: I would be for writing "valid" everywhere applicable, as it makes the spec less brittle. E.g. someone may decide later on to add some VUs to VkRect2D too.

We could alternatively just state that "must: be an X" where the parameter is readable implies it must be a valid X? Only non-const pointers are written, so should be easy enough to write something sane up. I'd be ok with that.